junit/src/main/java/cucumber/api/junit/Cucumber.java

```diff
@@ -91,6 +93,14 @@ protected Description describeChild(FeatureRunner child) {
     }
 
     @Override
+    public void filter(Filter filter) throws NoTestsRemainException {
+        // dont pass filter to children as ParentRunner does by default
```

Comment:
This won't play well with the rerun functionality in surefire which uses filters to prune the successful tests. By not passing the filter on to the children these will not be pruned. Some more sophistication is needed.